piwheels/master/the_scribe.py

```diff
@@ -312,46 +318,91 @@ def write_project_page(self, package):
         :param str package:
             The name of the package to write the project page for
         """
-        versions = sorted(
-            self.db.get_project_versions(package),
-            key=lambda row: pkg_resources.parse_version(row.version),
-            reverse=True)
-        files = sorted(
-            self.db.get_project_files(package),
-            key=lambda row: (
-                pkg_resources.parse_version(row.version),
-                row.filename
-            ), reverse=True)
-        if files:
-            dependencies = self.db.get_file_dependencies(files[0].filename)
-            if 'apt' in dependencies:
-                dependencies['apt'] = dependencies['apt'] - self.default_libs
-        else:
-            dependencies = {}
         self.logger.info('writing project page for %s', package)
         pkg_dir = self.output_path / 'project' / package
         mkdir_override_symlink(pkg_dir)
-        dt = datetime.now()
         with AtomicReplaceFile(pkg_dir / 'index.html', encoding='utf-8') as index:
             index.file.write(self.templates['project'](
                 layout=self.templates['layout']['layout'],
-                package=package,
-                versions=versions,
-                files=files,
-                dependencies=dependencies,
-                format_size=format_size,
-                timestamp=dt.strftime('%Y-%m-%d %H:%M'),
-                url=lambda filename, filehash:
-                    '/simple/{package}/{filename}#sha256={filehash}'.format(
-                        package=package, filename=filename, filehash=filehash),
-                page='project'))
+                page='project',
+                package=package))
         try:
             # See write_package_index for explanation...
             canon_dir = pkg_dir.with_name(canonicalize_name(pkg_dir.name))
             canon_dir.symlink_to(pkg_dir.name)
         except FileExistsError:
             pass
 
+    def write_json_file(self, package):
+        """
+        (Re)writes the JSON file for the specified package.
+
+        :param str package:
+            The name of the package to write the JSON file for
+        """
+        self.logger.info('writing json file for %s', package)
+        pkg_dir = self.output_path / 'project' / package / 'json'
```

Comment:
This line works?!?  I would have thought __pkg_dir = "/".join((self.output_path, 'project', package, 'json'))__

Same for line 347

Reply:
:)

Pathlib overloads the `/` operator to allow this: https://docs.python.org/3/library/pathlib.html

Reply:
Super cool!!  Thanks!

Reply:
I do like the pathlib syntax; it's also entirely "obvious" to the reader what it's doing (may not be obvious at first that it actually works, but at least the intent is entirely clear :)